base/libgit2/types.jl

```diff
@@ -494,6 +502,15 @@ for (typ, reporef, sup, cname) in [
     end
 end
 
+## Calling `GitObject(repo, ...)` will automatically resolve to the appropriate type.
+function GitObject(repo::GitRepo, ptr::Ptr{Void})
+    T = objtype(ccall((:git_object_type, :libgit2), Consts.OBJECT, (Ptr{Void},), ptr))
+    T(repo, ptr)
+end
+
+
+
+
```

Comment:
Four newlines seems like a lot of vertical whitespace. Perhaps for consistency it should be 1-2?